spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/DefaultDatastoreEntityConverter.java

```diff
@@ -79,11 +81,23 @@ public ReadWriteConversions getConversions() {
 
 	@Override
 	public <T, R> Map<T, R> readAsMap(Class<T> keyType, TypeInformation<R> componentType,
-			BaseEntity entity) {
+			BaseEntity entity, TypeInformation typeInformation) {
 		if (entity == null) {
 			return null;
 		}
-		Map<T, R> result = new HashMap<>();
+		Object customMap = null;
+		if (typeInformation != null && !typeInformation.getType().isInterface()) {
+			try {
+				customMap = ((Constructor<?>) typeInformation.getType().getConstructor()).newInstance();
```

Comment:
This just doesn't seem right. Can we put the burden on the caller to create the map?

Reply:
Could you elaborate why it doesn't seem right?

Reply:
I think if the caller has to create it, it will perhaps be inconsistent with the other cases.